src/App.tsx

```diff
@@ -1,61 +1,148 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoFromServer } from './types/TodoFromServer';
+import { TodoList } from './components/TodoList';
+import { getTodos } from './utils/getTodos';
+import { addTodo } from './utils/addTodo';
+
+const pattern = /[А-ЯёїіA-Z\d\s]/i;
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState('0');
+  const [todos, setTodos] = useState(getTodos(todosFromServer));
+  const [showErrorTitle, setShowErrorTitle] = useState(false);
+  const [showErrorUser, setShowErrorUser] = useState(false);
+
+  function handleChange(
+    func: React.Dispatch<React.SetStateAction<string>>,
+    event: React.ChangeEvent<HTMLInputElement>
+    | React.ChangeEvent<HTMLSelectElement>,
+  ): void {
+    const { value } = event.target;
+    const lastIndex = value.length - 1;
+
+    if (value.length && pattern.test(value[lastIndex])) {
+      func(value);
+    }
+
+    if (!event.target.value) {
+      func(value);
+    }
+  }
```

Comment:
don't add a common function for onChange. 
just use separate functions: 
handleTitle/input/select = (event: React.ChangeEvent<HTMLInputElement>) {
setTitle/input/select..
}